core/node/storage/pg_stream_store.go

```diff
@@ -42,6 +45,61 @@ func GetRiverNodeDbMigrationSchemaFS() *embed.FS {
 	return &migrationsDir
 }
 
+type txnFn func(ctx context.Context, tx pgx.Tx) error
+
+func createSettingsTable(store *PostgresStreamStore, partitions int) txnFn {
```

Comment:
It seems like it would be more consistent to pass `PostgresStreamStore` as "this"?
```suggestion
func (s *PostgresStreamStore) createSettingsTableFn(partitions int) txnFn {
```

Reply:
If I can do that? I wasn't aware of how go methods work under the hood, I was using this to pass into the underlying event store for running before migrations are applied.